tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,46 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_eviction_propagation(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(proactor_threads=1, cache_mode="true", maxmemory="290mb")
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    seeder = df_seeder_factory.create(port=master.port, val_size=1000, stop_on_failure=False)
+
+    await c_master.execute_command("DEBUG POPULATE 6200000")
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica, timeout=3000)
+
+    # Check data after full sync
+    await check_all_replicas_finished([c_replica], c_master)
+    await check_data(seeder, [replica], [c_replica])
+
+    # Stream more data in stable state
+    await seeder.run(target_ops=10000000, target_deviation=0.0001)
+
+    info = await c_master.info("stats")
+    assert info["evicted_keys"] > 0, "Weak testcase. No eviction was triggered"
+
+    while True:
+        info = await c_master.info("stats")
+        evicted_1 = info["evicted_keys"]
+        time.sleep(2)
+        info = await c_master.info("stats")
+        evicted_2 = info["evicted_keys"]
+        if evicted_2 == evicted_1:
+            break
+        else:
+            print("waiting for eviction to finish...", end="\r", flush=True)
+
+    # Check data after stable state stream
+    await check_all_replicas_finished([c_replica], c_master)
+    await check_data(seeder, [replica], [c_replica])
```

Comment:
this will only check the keys the seeder sent, but what if we evict keys that were added by debug populate? we dont check they are in master/replica

Reply:
I made sure we don't evict any keys when we do debug populate. And it doesnt matter any way because, we only do replicaof after debug population to master is finished. as I mentioned in the chat, debug populate's insertion can't be replicated to replica right now. This is still a todo for debug populate.

Reply:
I dont understand. We do not evict while running debug populate we evict when the seeder is running but when we evict when the seeder is running we can evict keys inserted by debug populate.
We want to check that when evicting keys we send the del to the replica and the replica deletes them, it could be keys inserted by seeder it could be keys inserted by debug populate

Reply:
yes, that's correct. here the check data only compares the data added by the seeder.